community/modules/scripts/wait-for-startup/main.tf

```diff
@@ -13,22 +13,40 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+locals {
+  instance_names = toset(compact(concat(var.instance_names, [var.instance_name])))
+}
+
 data "google_compute_instance" "vm_instance" {
-  name    = var.instance_name
+  for_each = local.instance_names
```

Comment:
IIUC, `for_each` can't be used with values not known in advance.
If we one uses `random_pet` or `random_id` modules to come up with `instance_names`, this line will throw.

Reply:
One way to fix it is to not use `toset(compact(concat(...))))` that has an unknown length,
but raw `var.instance_names` itself that has a better chances of having known length.

Reply:
The `for_each` variable can only be assigned a `set()` value, otherwise it throws an error:

```
Error: Invalid for_each argument

  on modules/embedded/community/modules/scripts/wait-for-startup/main.tf line 22, in data "google_compute_instance" "vm_instance":
  22:   for_each = local.instance_names
    ├────────────────
    │ local.instance_names is tuple with 5 elements

The given "for_each" argument value is unsuitable: the "for_each" argument
must be a map, or set of strings, and you have provided a value of type
tuple.
```

Are there any other ways of preventing the possibility of an unknown length of instance names?  I can't think of any, but I'm certainly not a Terraform expert.

Reply:
It looks like `count` doesn't have the same restriction as `for_each`: https://github.com/hashicorp/terraform/issues/30039.  I will use `count` instead.